store/postgres/src/relational/index.rs

```diff
@@ -558,6 +729,115 @@ impl CreateIndex {
     }
 }
 
+#[derive(Debug)]
+pub struct IndexList {
+    pub(crate) indexes: HashMap<String, Vec<CreateIndex>>,
+}
+
+impl IndexList {
+    pub fn load(
+        conn: &mut PgConnection,
+        site: Arc<Site>,
+        store: DeploymentStore,
+    ) -> Result<Self, StoreError> {
+        let mut list = IndexList {
+            indexes: HashMap::new(),
+        };
+        let schema_name = site.namespace.clone();
+        let layout = store.layout(conn, site)?;
+        for (_, table) in &layout.tables {
+            let table_name = table.name.as_str();
+            let indexes = catalog::indexes_for_table(conn, schema_name.as_str(), table_name)?;
+            let collect: Vec<CreateIndex> = indexes.into_iter().map(CreateIndex::parse).collect();
+            list.indexes.insert(table_name.to_string(), collect);
+        }
+        Ok(list)
+    }
+
+    pub fn indexes_for_table(
+        &self,
+        namespace: &Namespace,
+        table_name: &String,
+        dest_table: &Table,
+        postponed: bool,
+        concurent_if_not_exist: bool,
+    ) -> Result<Vec<(Option<String>, String)>, Error> {
+        let mut arr = vec![];
+        if let Some(vec) = self.indexes.get(table_name) {
+            for ci in vec {
+                if ci.fields_exist_in_dest(dest_table)
+                    && !ci.is_default_non_attr_index()
+                    && !(ci.is_id() && dest_table.immutable)
+                    && postponed == ci.to_postpone()
```

Comment:
Small suggestion, a comment here explaining why all this conditions is neccessary would be great

Reply:
addressed